hail/src/main/scala/is/hail/expr/ir/InferPType.scala

```diff
@@ -6,9 +6,13 @@ import is.hail.utils._
 
 object InferPType {
 
-  def clearPTypes(x: IR): Unit = {
-    x._pType = null
-    x.children.foreach { c => clearPTypes(c.asInstanceOf[IR]) }
+  def clearPTypes(x: BaseIR): Unit = {
```

Comment:
why is this necessary? When we have a ptype-d IR, it must contain only IR, not BaseIR.

Reply:
So where I run (inside TestUtils.assertEvalsTo), there can be relational nodes.  I have to do this because we run multiple compile strategies now (opt and non-opt).  Those IR can be shared arbitrarily with IR that get ptypes assigned by InferPTypes.  InferPTypes asserts no ptypes are already assigned, so they all need to be cleared.

Reply:
got it, sure.